### PR TITLE
Feature/complete ap01 processing

### PR DESCRIPTION
**Pull request by danfalc**

### CODE REVIEW

File: conf/env.conf [hunk 1]
```diff
@@ -1,11 +1,11 @@
 testing {
```
[the remark this answers is not in the document]
Comment (danfalc) [reply conf/env.conf:1]: Ah ffs

File: src/test/java/uk/gov/companieshouse/data/datamodel/Director.java [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import uk.gov.companieshouse.utils.DateFormat;
+
+
+public class Director {
+
+    private final String dateOfAppointmentField;
```
Comment (rsuller) [inline src/test/java/uk/gov/companieshouse/data/datamodel/Director.java:14]: Does a date of appointment really need to be part of the Director? - Surely that is just the date it is submitted/processed.

File: src/test/java/uk/gov/companieshouse/data/datamodel/Director.java [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+    private final String titleOtherField;
+    private final String partialDobField;
+    private final String forenameField;
+    private final String middleNamesField;
+    private final String surnameField;
+    private final String countryUsuallyResidentField;
+    private final String dobField;
+    private final String countryField;
+    private final String countryOtherField;
+    private final String uraBarcodeField;
```
Comment (rsuller) [inline src/test/java/uk/gov/companieshouse/data/datamodel/Director.java:25]: Barcode. Is that something that needs to be stored in the director's details? Again, isn't it just an internal processing thing?
Or is it unique to the Director?

File: src/test/java/uk/gov/companieshouse/data/datamodel/Director.java [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+    private boolean uraBarcodeRequired;
+    private boolean noChangeToUra;
+    private final List<String> nationalityField;
+    private final String occupationField;
+    private final Address serviceAddressField;
+    private final String  serviceAddressHouseNumberField;
```
Comment (rsuller) [inline src/test/java/uk/gov/companieshouse/data/datamodel/Director.java:31]: Does the service number bit need to be separate? Wouldn't it be part of the service address?

File: src/test/java/uk/gov/companieshouse/stepdefinitions/CommonStepDefs.java [hunk 5]
```diff
@@ -81,7 +81,7 @@ public void companyHistoryInformationIsUpdated(String formName) {
         companyDetailsScreen
                 .waitUntilDisplayed()
                 .getExpectedTransactionFromHistory(form.getType(), documentDetails.getReceivedDate(), "ACCEPTED",
-                        form.getTransactionHistoryDescription(), "WEB LOGIC");
+                        form.gettransactionHistoryPartialDescription());
```
Comment (rsuller) [inline src/test/java/uk/gov/companieshouse/stepdefinitions/CommonStepDefs.java:84]: `getTransactionHistoryPartialDescription()`

File: src/test/java/uk/gov/companieshouse/data/datamodel/Director.java (hunk 2 above)
Comment (danfalc) [reply src/test/java/uk/gov/companieshouse/data/datamodel/Director.java:14]: Good [point. Removed

File: src/test/java/uk/gov/companieshouse/data/datamodel/Director.java (hunk 3 above)
Comment (danfalc) [reply src/test/java/uk/gov/companieshouse/data/datamodel/Director.java:25]: This is the URA barcode we were talking about that is required for some forms. Specific to a director

File: src/test/java/uk/gov/companieshouse/data/datamodel/Director.java [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+            return this;
+        }
+
+        public Director build() {
+            return new Director(this);
+        }
+
+    }
+
+    public static void main(String[] args) {
```
[the remark this answers is not in the document]
Comment (danfalc) [reply src/test/java/uk/gov/companieshouse/data/datamodel/Director.java:288]: removed

File: src/test/java/uk/gov/companieshouse/stepdefinitions/CommonStepDefs.java (hunk 5 above)
Comment (danfalc) [reply src/test/java/uk/gov/companieshouse/stepdefinitions/CommonStepDefs.java:84]: changed

File: src/test/java/uk/gov/companieshouse/data/datamodel/Director.java (hunk 4 above)
Comment (danfalc) [reply src/test/java/uk/gov/companieshouse/data/datamodel/Director.java:31]: Removed, along with the postcode and residential address examples

**Review by rsuller — APPROVED** [review]

**Pull request merged**